### PR TITLE
Specs for renewal dependent add

### PR DESCRIPTION
**Pull request by krbhavith**

### CODE REVIEW

**Comment by brianweiner** [comment]
Regarding #publish test failures

https://github.com/dchbx/gluedb/blob/release-4.2/app/models/enrollment_action/renewal_dependent_add.rb#L15

#added_dependents makes that call on L15 which utilizes RenewalComparisonHelper. This is included into the class so you need to mock the request at that level. 

https://github.com/dchbx/gluedb/pull/73/files#diff-78fc4216cf711aac3433c87c4ee48b1cR135 In this example the subject is the class itself so I only had to mock the subject request.

In your case you need to allow subject.class to receive a message ':same_carrier_renewal_candidates' with an action as an attribute. It should respond with an array of policies (I think) that should have enrollees which is an array of elements that have a m_id attribute.

Basically it's saying my current action includes members [1,2,3] does this policy have enrollees with m_ids that represent an extra id from what we started with.

File: spec/models/enrollment_action/renewal_dependent_add_spec.rb [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  before do
+    allow(subject).to receive(:same_carrier_renewal_candidates).with(event).and_return([member])
+    allow(subject).to receive(:renewal_dependents_added?).with(member, event).and_return(true)
+  end
+  it "qualifies" do
+    expect(subject.qualifies?([event])).to be_truthy
+  end
+end
+
+describe EnrollmentAction::RenewalDependentAdd, "given an enrollment event set that:
```
Comment (brianweiner) [inline spec/models/enrollment_action/renewal_dependent_add_spec.rb]: L26 -> L46 seems to be an accidental duplicate of the next block

File: spec/models/enrollment_action/renewal_dependent_add_spec.rb [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  let(:primary_db_record) { instance_double(ExternalEvents::ExternalMember, :persist => true) }
+
+  let(:new_policy_cv) { instance_double(Openhbx::Cv2::Policy, :enrollees => [enrollee_primary]) }
+
+  let(:renewal_dependent_add_event) { instance_double(
+    ::ExternalEvents::EnrollmentEventNotification,
+    :all_member_ids => [1,2],
+    :event_responder => event_responder,
+    :event_xml => event_xml,
+    :is_shop? => true,
```
Comment (brianweiner) [inline spec/models/enrollment_action/renewal_dependent_add_spec.rb]: is_shop? doesn't get used in this class, so for clarity I would avoid setting a value for it within the double.

I find it easier to understand when I only set values for the double that I expect to be called within the relevant method.
